Pedal Controlled Proportional Assist

tttjm1 opened the pull request:
//Pedal Controlled Proportional Assist.
//Automatic assist/regen using pedal position and engine signals
//(No joystick required)
//requires power user cable from Licontrol to vehicle ECU signals

**requires @Hurricos PR3 to Measure vehicle speed using pin-change IRQ on VSS**

Configured for:
48S & 60S FoMoCo / zero current hack / VOLTAGE_SPOOFING_LINEAR

Tested on a 2001 manual, consumes 82-15% SOC in ~60miles at approximately 150mpg(imp) / 120 mpg US

tttjm1 committed:
Update operatingModes.cpp

add mode_proportional_auto_assist

tttjm1 committed:
Update config.h

add mode_proportional_auto_assist();

tttjm1 committed:
refined light assist/regen

doppelhub reviewed (commented):
Please document that this mode requires the Power User Cable, which is technically an optional installation step.  Ideally also add safeguards to prevent this mode from working until LiControl actually detects that VSS and TPS are connected.

I will test this code out tomorrow.  My goal is to pull all the open requests into Prerelease soon, and then into Main.

tttjm1 commented:
Comment was regarding Licontrol 12v supply failure, now known to be unrelated to this code.

tttjm1 committed:
Update operatingModes.cpp

response now linear rather than exponential.

tttjm1 committed:
Update operatingModes.cpp

defined integers at start of code

tttjm1 committed:
Update operatingModes.h

tttjm1 committed:
Update operatingModes.cpp

tttjm1 committed:
Update operatingModes.cpp

tttjm1 committed:
Update operatingModes.h

add definitions

tttjm1 commented:
Updated with better coding and refinements.

tttjm1 committed:
improved driveability

optimised assist levels for smooth operation.
replaced OEM brake assist entirely to give consistent regen levels at a set speed in all gears.

tttjm1 committed:
Update operatingModes.h

tttjm1 commented:
Refined to provide better far better driveability.

tttjm1 committed:
Update operatingModes.cpp

tttjm1 committed:
Update operatingModes.cpp

latest refinement to driveability

tttjm1 committed:
Update operatingModes.h

latest driveability refinement.

tttjm1 committed:
Update operatingModes.cpp

tttjm1 committed:
Update operatingModes.h

tttjm1 commented:
Major revision results in a seamless regen/assist operation.

Assist is proportional to accelerator pedal position, vehicle speed and engine load
Regen is enable by braking and is proportional to vehicle speed and engine speed.

tttjm1 committed:
Update operatingModes.h

tttjm1 committed:
Update operatingModes.cpp

tttjm1 commented:
Added factor to fade regen to virtually zero as 1000rpm is reached.
(this prevents stalling whilst braking and changing gear)

tttjm1 committed:
Update operatingModes.h

reduced brake regen by 10% - avoids any "overflow" cut out

tttjm1 committed:
Update operatingModes.cpp

add hysteresis to assist level, prevents P1440 with rapid changes in throttle position.

tttjm1 commented:
add hysteresis to assist level, this prevents P1440 which previously occurred during rapid changes in throttle position. No change in actual vehicle performance.

tttjm1 committed:
Update operatingModes.cpp

tttjm1 committed:
Update operatingModes.cpp

Add upper limit to assist demand (90%)

tttjm1 committed:
Add hysterisis to MAP signal

Adding hysteresis to the MAP signal makes the assist buttery smooth. Without this change it was still possible to trigger P0420 with aggressive throttle inputs at the higher currents when as a result of low SOC.

tttjm1 commented:
Add hysteresis to MAP signal

Adding hysteresis to the MAP signal makes the assist buttery smooth. Without this change it was still possible to trigger P1440 with aggressive throttle inputs at the higher currents when as a result of low SOC.
